astropy/time/core.py

```diff
@@ -817,6 +863,79 @@ def masked(self):
     def mask(self):
         return self._time.mask
 
+    def insert(self, obj, values, axis=0):
+        """
+        Insert values before the given indices in the column and return
+        a new `~astropy.time.Time` or  `~astropy.time.TimeDelta` object.
+
+        The values to be inserted must conform to the rules for in-place setting
+        of ``Time`` objects (see ``Get and set values`` in the ``Time``
+        documentation).
+
+        The API signature matches the `np.insert` API, but is more limited.
+        The specification of insert index ``obj`` must be a single integer,
+        and the ``axis`` must be ``0`` for simple row insertion before the
+        index.
+
+        Parameters
+        ----------
+        obj : int
+            Integer index before which ``values`` is inserted.
+        values : array_like
+            Value(s) to insert.  If the type of ``values`` is different
+            from that of quantity, ``values`` is converted to the matching type.
+        axis : int, optional
+            Axis along which to insert ``values``.  Default is 0, which is the
+            only allowed value and will insert a row.
+
+        Returns
+        -------
+        out : `~astropy.time.Time` subclass
+            New time object with inserted value(s)
+
+        """
+        # Validate inputs: obj arg is integer, axis=0, self is not a scalar, and
+        # input index is in bounds.
+        if not isinstance(obj, (int, np.integer)):
```

Comment:
The secure, duck-typing way to get an index is
```
idx0 = operator.index(obj)
```
(can be within `try/except` if you want a different error message)